src/index.ts

```diff
@@ -179,7 +180,7 @@ async function onError(instance: AxiosInstance, error: AxiosError) {
 	config.retry = typeof config.retry === 'number' ? config.retry : 3;
 	config.retryDelay =
 		typeof config.retryDelay === 'number' ? config.retryDelay : 100;
-	config.instance = config.instance || instance;
```

Comment:
A possible regression, if anyone uses `config.instance` in a callback (like `onRetryAttempt`) *without* filling it, and expects it to be `axios`. Do you believe this is a valid case, or is it safe to just ignore it?

If you prefer, this can be reverted, and I'll just leave the deprecation note.